catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java

```diff
@@ -1200,6 +1244,27 @@ public static String getValueString(Collection<Serializable> collection) {
                 .collect(Collectors.joining(", "));
     }
 
+    private byte[] getThumbnail(String thumbnailUrlStr) {
+        byte[] thumbnail = null;
+
+        try {
+            URI thumbnailURI = new URI(thumbnailUrlStr);
+            ResourceResponse resourceResponse = null;
+            try {
+                resourceResponse = resourceReader.retrieveResource(thumbnailURI, new HashMap<>());
```

Comment:
Is this the only place where the `resourceReader` is being used in the DAGConverter?  I saw that this is set in the constructor, and NSILISource is creating a new DAGConverter for each DAG in the source response.  If there are 100 DAGs in the response, 100 DAGonverters are created.  Can we avoid this?